### DISCUSSION TITLE
Some contents in pdf page is not identifying either as text or image

### DISCUSSION THREAD

**Discussion by UdayaKUnnikrishnan** [pymupdf/PyMuPDF#1876] [Looking for help] [answered]
Pdf' page contain text that can not be copied. Such text regions is not identified as image or text by pymupdf. 
Please help me how can I  extract such contents from pdf page.

**Comment by JorjMcKie** [comment, marked as answer]
You must provide an example - otherwise I cannot help.
As is typical for PDF, there are a number of possible explanations. Among them:
You can use so-called "line art" to simulate text - like a capital "D" can be drawn by a line "|" and a left-open semi-circle.
Line art is neither text nor image ...

**Reply by UdayaKUnnikrishnan** [reply]
Thank you for the quick reply. I could not share the pdf as it is confidential. I used page.get_bboxlog() to get the list of rectangles in the page and created a sample pdf using this info . I am attaching the pdf created using these rectangle info. 
[sample.pdf](https://github.com/pymupdf/PyMuPDF/files/9349833/sample.pdf)

Text present in the fill-path rectangles( marked with red line)  can not be extracted. I need to extract text from these regions

**Reply by JorjMcKie** [reply]
Well done! "Fill path" and "stroke path" are areas where there exists line art.
So my assumption seems to be correct: it is text simulated by line art.
While you certainly can extract those drawings themselves, you will not be able to interpret it as text of course.
This means, your only option is OCR-ing the page and **_then_** extract the recognized text.
PyMuPDF support OCR via an installed Tesseract software - presumably you know that and how to get it going.

**Reply by JorjMcKie** [reply]
Or install / use [ocrmypdf](https://pypi.org/project/ocrmypdf/), OCR the file, then process its output PDF as usual.

**Reply by UdayaKUnnikrishnan** [reply]
Thank you so much for your support. Yes I know about ocr support in PyMuPDF.
